docs/interfaces/action-service.txt

```diff
@@ -1,27 +1,19 @@
 # Action Service Interface
 
-// TODO: More Exceptions?
-Exception ActionNotFound(name: string);
+Exception ActionException(message: string);
 
 
 /**
  * Executes multiple actions in the conext of the user given by the user_id.
+ * All actions are successful if no exception is thrown, else no action was
+ * processed.
```

Comment:
I thought the event store does not provide the possibility that different actions are sent together and may handled independently.

Email is a really special case:

If the request user asks me to send three mails I would try to send them all. If all were successful (means just successfully handled to the mail transfer agent), I can try to save the email_sent timestamp to the user model. If this fails ... bad luck. If not ... ok.

If the first email is successful and the second not, what should I do? Should I try to save the timestamp? Should I first tra to sent the third email? I do not have different7independent workers or threads so we must decide the order of what to do.

Reply:
> I thought the event store does not provide the possibility that different actions are sent together and may handled independently.

It does! You just have to change your execution (during the dispatching): Instead of collectiong all events from all actions and sending them once, send the events after each action.

The order is given by the client: all actions are evaluated in the order given. There are more cases: You can select all users to send an email, but some users doesn't even have an email in the system. This is not limited to users: Most bulk actions may send multiple models, for which the action is not allowed. This is a huge overhead in the client to query and filter, but easily done, if the actions are performed independent of eachother.

Reply:
Changed.

Reply:
I want this to change back, see my email.